src/lib/y2partitioner/dialogs/blk_device_resize.rb

```diff
@@ -94,6 +94,7 @@ def next_handler
 
       def detect_space_info
         return unless formatted? && controller.committed_device? && !swap?
+        return unless controller.committed_current_filesystem?
```

Comment:
Why did you remove the `!swap?` part? As far as I understand the previous code, `Filesystem#detect_space_info` should not be called for swap filesystems. But that check is gone now.

Reply:
I didn't. That's an extra line of code. Otherwise the entire expression would have been too long for Rubocop's liking.

Reply:
Right. I got confused because then the `formatted? && controller.committed_device?` part is absolutely redundant with the condition you added. Isn't it? This should be enough and more clear:

```ruby
def detect_space_info
  return unless controller.comitted_current_filesystem? && !swap?
  # the rest
end
```

Reply:
I am not sure. I did the defensive approach. Can't there be other situations where only the old condition is true?

Reply:
I'm pretty sure this is totally sure.

First, semantically:

- If `formatted?` is false, then there is no filesystem on the device. Thus `committed_current_filesystem?` MUST be false as well.
- If `controller.committed_device?` is false, then the device was not in the original devicegraph. Thus, its filesystem was not there either.

And now looking at the code:

- If `formatted?` is false, then `committed_current_filesystem?` is also false because that method starts with `return false unless device.formatted?`
- If `controller.committed_device?` is false, then `committed_current_filesystem?` is also false because it directly relies on `committed_filesystem` whose first line is `return nil unless committed_device?`.

Reply:
OK then. Removing the extra conditions.